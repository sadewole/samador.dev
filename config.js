module.exports = {
  url: "https://samador-dev.netlify.app/",
  pathPrefix: "/",
  title: "Samador",
  description:
    "My personal portfolio. This project is built along with blog for sharing my thoughts.",
  copyright: "Â© All rights reserved.",
  postsPerPage: 4,
  menu: [
    {
      label: "Home",
      path: "/",
    },
    {
      label: "About me",
      path: "/about",
    },
    {
      label: "Blogs",
      path: "/blogs",
    },
  ],
  author: {
    name: "Samuel Adewole",
    photo: "/images/photo2.jpg",
    bio: "Software engineer | Creator of all things possible | Achiever",
    contacts: {
      email: "samueladewole15@gmail.com",
      facebook: "",
      telegram: "",
      twitter: "samador9",
      github: "sadewole",
      rss: "",
      vkontakte: "",
      linkedin: "samador9",
      instagram: "",
      line: "",
      gitlab: "",
      weibo: "",
      codepen: "",
      youtube: "",
      soundcloud: "",
      medium: "",
    },
  },
}
