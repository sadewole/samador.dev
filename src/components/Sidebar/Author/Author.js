import React from "react"
import { withPrefix, Link } from "gatsby"

const Author = ({ author }) => {
  return (
    <div className="mb-4">
      <Link to="/">
        <img
          src={withPrefix(author.photo)}
          className="rounded"
          width="75"
          height="75"
          alt={author.name}
        />
      </Link>
      (
      <h2 className="text-3xl mb-2 leading-normal font-semibold text-white">
        <Link className="" to="/">
          {author.name}
        </Link>
      </h2>
      <p className="text-lg font-light leading-relaxed mt-0 mb-4 text-gray-300">
        {author.bio}
      </p>
    </div>
  )
}

export default Author
