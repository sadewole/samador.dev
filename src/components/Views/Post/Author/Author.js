import React from "react"
import PropTypes from "prop-types"
import { withPrefix } from "gatsby"
import { getContact } from "../../../../utils"

const Author = ({ author }) => {
  return (
    <div className="my-5 flex items-center">
      <img
        src={withPrefix(author.photo)}
        width="60"
        height="60"
        className="rounded mr-3"
        alt={author.name}
      />
      <div>
        <p>
          Written by{" "}
          <a
            className=""
            href={getContact("twitter", author.contacts.twitter)}
            rel="noopener noreferrer"
            target="_blank"
          >
            <strong>{author.name}</strong>
          </a>{" "}
        </p>
        <p className="text-sm font-light leading-relaxed mt-0 mb-4 text-gray-300">
          {author.bio}.
        </p>
      </div>
    </div>
  )
}

Author.propTypes = {
  author: PropTypes.object,
}

export default Author
