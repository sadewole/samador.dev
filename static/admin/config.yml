backend:
  name: git-gateway
  branch: pages-cofig

media_folder: "static/media"
public_folder: "/media"

collections:
  - name: "blog"
    label: "Blog"
    folder: "content/blog"
    create: true
    slug: "{{year}}-{{month}}-{{day}}-{{slug}}"
    fields:
      - { label: "Layout", name: "layout", widget: "hidden", default: "blog" }
      - { label: "Title", name: "title", widget: "string" }
      - { label: "Slug", name: "slug", widget: "string" }
      - { label: "Publish Date", name: "date", widget: "datetime" }
      - { label: "Body", name: "body", widget: "markdown" }
      - { label: "Tags", name: "tags", widget: "list" }

  - name: "pages"
    label: "Pages"
    folder: "content/pages"
    create: true
    slug: "{{slug}}"
    fields:
      - { label: "Layout", name: "layout", widget: "hidden", default: "page" }
      - { label: "Title", name: "title", widget: "string" }
      - { label: "Slug", name: "slug", widget: "string" }
      - { label: "Draft", name: "draft", widget: "boolean", default: true }
      - { label: "Body", name: "body", widget: "markdown" }
