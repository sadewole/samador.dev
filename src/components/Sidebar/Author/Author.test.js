import React from "react"
import { render } from "@testing-library/react"
import renderer from "react-test-renderer"

import Author from "./Author"
import { author } from "../../../../config"

describe("Author", () => {
  const props = {
    author: {
      name: author.name,
      photo: "/photo.jpg",
      bio: "test",
    },
  }

  test("must be author name", () => {
    const { getByTestId } = render(<Author {...props} />)
    expect(getByTestId("author")).toHaveTextContent("Samuel Adewole")
  })

  test("render correctly", () => {
    const tree = renderer.create(<Author {...props} />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
