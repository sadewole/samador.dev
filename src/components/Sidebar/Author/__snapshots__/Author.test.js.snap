// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Author render correctly 1`] = `
<div
  className="mb-4"
>
  <a
    href="/"
  >
    <img
      alt="Samuel Adewole"
      className="rounded transform hover:scale-105"
      height="85"
      src="/photo.jpg"
      width="85"
    />
  </a>
  <div />
  <h2
    className="text-2xl mb-2 leading-normal font-semibold text-white"
  >
    <a
      className=""
      data-testid="author"
      href="/"
    >
      Samuel Adewole
    </a>
  </h2>
  <p
    className="text-sm font-light leading-relaxed mt-0 mb-4 text-gray-300"
  >
    test
  </p>
</div>
`;
